File: backend/src/main/java/io/turismo/backend/model/Tourist.java
package io.turismo.backend.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.HashSet;
import java.util.Set;
import java.util.UUID;

@Entity
@Table(name = "tourists")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Tourist {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name="tourist_id")
    private UUID touristId;

    @Column(name = "birth_date")
    private LocalDate birthDate;

    @OneToOne
    @JoinColumn(name = "user_id", nullable = false, unique = true)
    private User user;

}

File: frontend/src/app/core/models/tourist.ts
import { UserCreate } from './user';

export interface TouristCreate {
  birthDate: string;
  user: UserCreate;
}

export interface Tourist {
  touristId: string;
  birthDate: string;
  userId: string;
  name: string;
  phone: string;
}

export interface TouristUpdate {
  birthDate?: string;
  name?: string;
  phone?: string;
  languages?: string[];
}
